src/IOConnection.cpp

```diff
@@ -40,14 +40,15 @@ namespace eipScanner {
 			, _originatorVendorId{0}
 			, _originatorSerialNumber{0}
 			, _serialNumber{0}
-			, _outputData()
+			, _outputData{nullptr}
+			, _lastHandleTime(std::chrono::steady_clock::now())
 			, _receiveDataHandle([](auto a, auto b, auto data) {})
 			, _closeHandle([]() {}) {
 	}
 
 	IOConnection::~IOConnection() = default;
 
-	void IOConnection::setDataToSend(const std::vector<uint8_t> &data) {
+	void IOConnection::setDataToSend(std::vector<uint8_t> *data) {
```

Comment:
Here you have broken the compatibility and the users might get an unexpected compilation error. Method setDataToSend received a const reference and no additional copy happened. Please, return the previous version.

Reply:
How do you modify output assembly during processing? 
Are you calling setDataToSend every time you need to refresh the vector with full copy?

No problem to remove this commit from this PR as it's independant.

Reply:
Oh.. now I see what you mean. Yes, we copy the data into the internal buffer each time when it changes. For big vectors it is inefficient and your solution is better. However, I don't want to break the compatibility. 

I think we can make some workaround providing getter without 'const':

```cpp
std::vector<uint8_t>& IOConnection::getDataToSend() { return _outputData; }
```
A user can use it to mutate data without copping. It is a bit dirty way in may case, but it will work.